### DISCUSSION TITLE
Endless hero upgrades for Endless mode ?

### DISCUSSION THREAD

**Discussion by Cwpute** [ochadenas/cpudefense#137] [Ideas]
Hey, it's me again with impossibly complex solution to a minor problem 👋

Reading again through the comments regarding #136 , i was thinking that maybe hero upgrades could automatically scale up without limitations to accomodate for players getting into higher levels. This would reduce having to fine-tune each and every upgrade level for every hero everytime something wouldn't click right, as well as increase the possible play time in Endless quite tremendously.

For example, hero upgrades that reduce or increase a percentage would have a starting value and max/min value towards which each new upgrade would tend, with a basic predefined value for the first upgrade that would evolve as to never cross that target max/min. Upgrade price would also scale up, my guess would be to simply bump up the price of 1 crypto for each level.

I should try to actually cough up an actual equation for that later on 🪚

**Comment by ochadenas** [comment]
Actually, the strengths of the heroes *are* determined by formulas. If you want the details, have a look at Hero.kt starting from line 540. 

So in principle heroes can be defined for arbitrary levels. However, there are some where it doesn't make sense. We had the Tramiel discussion. Also Osborne is something like "6% per level", so this could eventually go above 100%. But apart from this, the heroes could effectively be made unlimited.

As for costs, it is really simple: Each upgrade costs as many coins as the number of the level (the one you want to upgrade it to).
